packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -3001,6 +3001,26 @@ describe('<Autocomplete />', () => {
     });
   });
 
+  describe('prop: renderOption', () => {
+    it('should pass getOptionLabel through ownerState in renderOption callback', () => {
+      const { getByRole } = render(
+        <Autocomplete
+          open
+          options={[{ label: 'one' }]}
+          getOptionLabel={() => `getOptionLabel`}
+          renderInput={(params) => <TextField {...params} autoFocus />}
+          renderOption={function f(...[option, , , ownerState]) {
+            return <li key="key">{ownerState.getOptionLabel(option)}</li>;
+          }}
+        />,
+      );
+
+      const listbox = getByRole('listbox');
+      const htmlOption = listbox.querySelector('li');
+      expect(htmlOption.innerHTML).to.equal('getOptionLabel');
```

Comment:
We should avoid DOM selectors like `innerHtml` in tests to avoid implementation details: 
```suggestion
      const renderedOption = screen.getByTestId('option');
      expect(renderedOption ).to.equal('Max');
```